src/app/components/register/register.component.ts

```diff
@@ -1,53 +1,71 @@
 import { RegisterRequest } from './../../register-request';
 import { CommonModule } from '@angular/common';
-import { Component } from '@angular/core';
-import {FormBuilder, FormControl, FormGroup, ReactiveFormsModule, Validators} from '@angular/forms'
+import { Component, inject } from '@angular/core';
+import {
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
 import { AuthService } from '../../services/auth.service';
 import { Router } from '@angular/router';
+import { ToastrService } from 'ngx-toastr';
 
 @Component({
   selector: 'app-register',
   standalone: true,
   imports: [ReactiveFormsModule, CommonModule],
   templateUrl: './register.component.html',
-  styleUrl: './register.component.scss'
+  styleUrl: './register.component.scss',
 })
 export class RegisterComponent {
+  toaster = inject(ToastrService);
+
+  email: FormControl<string | null> = new FormControl<string>('', [
+    Validators.required,
+    Validators.email,
+  ]);
+  password: FormControl<string | null> = new FormControl<string>('', [
+    Validators.required,
+    Validators.minLength(5),
+  ]);
 
-  email : FormControl<string | null> = new FormControl<string>('', [Validators.required, Validators.email]) 
-  password : FormControl<string | null> = new FormControl<string>('', [Validators.required, Validators.minLength(5)]) 
-  
   registerForm!: FormGroup;
 
```

Comment:
The RegisterForm can be assigned directly and email field can be declared directly in the form builder. there is no need to define it as a separate parameter and later assign it to the login form.

public  registerForm = this.formBuilder.group({
email: ['', [Validators.email, Validators.required]],
password: ['', Validators.required],
});

Reply:
code has been refactored